src/main/java/nextstep/subway/application/LineService.java

```diff
@@ -0,0 +1,66 @@
+package nextstep.subway.application;
+
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.dto.LineRequest;
+import nextstep.subway.dto.LineResponse;
+import nextstep.subway.dto.LineUpdateRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    @Autowired
+    private LineRepository lineRepository;
+
+    @Autowired
+    private StationService stationService;
+
+    public LineService(LineRepository lineRepository, StationService stationService) {
+        this.lineRepository = lineRepository;
+        this.stationService = stationService;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        Station upStation = stationService.getStationById(lineRequest.getUpStationId());
+        Station downStation = stationService.getStationById(lineRequest.getDownStationId());
+        Line persistLine = lineRepository.save(lineRequest.toLine(upStation, downStation));
+        return LineResponse.from(persistLine);
+    }
+
+    public List<LineResponse> findAllLines() {
+        List<Line> lines = lineRepository.findAll();
+        return lines.stream()
+                .map(line -> LineResponse.from(line))
```

Comment:
```suggestion
                .map(LineResponse::from)
```
메서드 참조를 사용할 수도 있을 것 같아요..!

Reply:
넵!